servers/physics_2d/body_2d_sw.cpp

```diff
@@ -468,7 +468,7 @@ void Body2DSW::integrate_forces(real_t p_step) {
 		linear_velocity = motion / p_step;
 
 		real_t rot = new_transform.get_rotation() - get_transform().get_rotation();
-		angular_velocity = rot / p_step;
+		angular_velocity = remainder(rot, 2.0 * Math_PI) / p_step;
```

Comment:
Why `2.0 * Math_PI` to wrap to `(-pi, pi)`? Also, prefer `Math_TAU` to `2.0 * Math_PI`.

Reply:
The remainder function takes the (signed) distance between the first parameter and the nearest integer multiple of the second value. By way of analogy, remainder is to rounding as the modulus is to truncation.

It may help to mentally step through all of the ranges for the angle. Between 0 and pi, the nearest integer multiple of 2*pi is 0, so the angle is unchanged. Between pi and 2*pi, the nearest integer multiple is 2pi, and the distance is `angle - 2*pi` (negative, because the angle is less than the nearest multiple), so the angle gets transformed to the range (-pi, 0), which is correct. Similarly, between -pi to 0, the nearest multiple is 0 and the angle is unchanged. Between -2pi and -pi, the nearest angle is -2pi and the distance is `2pi + angle`.

Reply:
Oh, I see. This behavior is correct in this situation, however, the name of this standard library function is very confusing, because this is not the typical behavior that the word "remainder" describes. A comment would be nice:

```suggestion
		// The remainder is relative to the nearest multiple rather than the multiple closest to zero.
		angular_velocity = remainder(rot, Math_TAU) / p_step;
```

Reply:
Arguments over the naming of the remainder function notwithstanding, I think we don't need to make a habit of describing functions from the standard library wherever they are used.

Perhaps it would be better to define an inline unroll_angle function to make it clearer the intent of this line, and that way the implementation could also be change with something more performant later if someone decided to do that. 

I also prefer 2.0*Math_PI in this case, because it makes the relationship to the range (-pi, pi) more explicit. 

That said as of right now this is the only place I'm aware of where angle unrolling is needed, so I'd lean against adding complexity and optimization prematurely. This is a bugfix PR, another PR could be used to redesign (and if I were doing that, I'd change to a stable angle-between-vectors algorithm instead of subtracting angles).

Reply:
We do have `fposmod`, which works for angles, but would give outputs on the range (0, TAU) if used with TAU or 2PI, so we would have to add a check with `> Math_PI` and then do `-= Math_TAU`.

Reply:
So is this good to merge as is, or should this be further documented/changed?

Reply:
I think it's good to merge as it is.